packages/@lwc/integration-karma/test/component/html-properties/index.spec.js

```diff
@@ -0,0 +1,250 @@
+        const setPropertyValue = (value) => {
+            // eslint-disable-next-line jest/valid-expect
+            let expected = expect(() => {
+                elm.setProperty(propName, value);
+            });
+            if (!readOnly) {
+                expected = expected.not; // only read-only logs an error here
+            }
+            expected.toLogErrorDev(
+                `Error: [LWC error]: The global HTML property \`${propName}\` is read-only.\n`
+            );
+        };
+
+        // Return a reasonable property value given what type the property accepts (boolean, number, keyword, etc.)
+        const getValueToSet = () => {
+            switch (propName) {
+                case 'contentEditable':
+                case 'draggable':
+                case 'hidden':
+                case 'spellcheck':
+                    return true;
+                case 'tabIndex':
+                    return 1;
+                case 'dir':
+                    return 'rtl';
+                case 'dropzone':
+                    return 'copy';
+                case 'inputMode':
+                    return 'numeric';
+                case 'lang':
+                    return 'en';
+                case 'style':
+                    return 'display: block;';
+                case 'translate':
+                    return 'yes';
+                default:
+                    return 'foo';
+            }
+        };
+
+        // Return the expected property value when no setter has been called
+        const getDefaultPropertyValue = () => {
+            switch (propName) {
+                case 'spellcheck':
+                    return document.createElement('div').spellcheck; // Firefox returns false, Chrome/Safari returns true
+                case 'draggable':
+                case 'hidden':
+                    return false;
+                case 'accessKey':
+                    return '';
+                case 'tabIndex':
+                    return -1;
+                case 'dir':
+                case 'id':
+                case 'lang':
+                    return '';
+                case 'role':
+                    return null;
+                case 'title':
+                    return '';
+                default:
+                    return undefined;
+            }
+        };
+
+        describe(propName, () => {
+            it('getter', () => {
+                const value = getPropertyValue();
+                expect(value).toEqual(getDefaultPropertyValue());
+            });
+
+            it('setter', () => {
+                const valueToSet = getValueToSet();
+                setPropertyValue(valueToSet);
+
+                const retrievedPropValue = getPropertyValue();
+                // In the case of non-original descriptors, the property value is not set
+                const expectedPropValue = isOriginalDescriptor ? valueToSet : undefined;
+                expect(retrievedPropValue).toEqual(expectedPropValue);
```

Comment:
It's not 100% clear to me why we have getters and setters that don't do anything other than log a warning. They seem to come from this code:

https://github.com/salesforce/lwc/blob/e9e5bc8a16fd54a3dd858ac6e646c46f1b45e5fe/packages/@lwc/engine-core/src/framework/restrictions.ts#L345-L365